spec/workers/moderator/sink_articles_worker_spec.rb

```diff
@@ -0,0 +1,34 @@
+require "rails_helper"
+
+RSpec.describe Moderator::SinkArticlesWorker, type: :worker do
+  include_examples "#enqueues_on_correct_queue", "medium_priority", 1
+
+  describe "#perform" do
+    it "returns early when user not found" do
+      expect { described_class.new.perform(nil) }.not_to raise_error
+      expect(described_class.new.perform(nil)).to be_nil
+    end
+
+    it "updates score for user's articles" do
+      article = create(:article, score: 20.0)
+      # flagging a user is the normal route to sink articles,
+      # and ensures score changes in an observable manner
+      create(:vomit_reaction, reactable: article.user)
+      allow(BlackBox).to receive(:article_hotness_score).and_call_original
+
+      described_class.new.perform(article.user_id)
+
+      expect(BlackBox).to have_received(:article_hotness_score)
+      expect(article.reload.score).not_to eq(20)
+    end
+
+    it "skips draft articles" do
+      article = create(:article, published: false)
+      allow(BlackBox).to receive(:article_hotness_score).and_call_original
+
+      described_class.new.perform(article.user_id)
+
+      expect(BlackBox).not_to have_received(:article_hotness_score)
```

Comment:
testing BlackBox seemed easier than stubbing the relation and expecting no instancce of Article to receive update_score (which is the layer this worker operates at).

If the implementation of Article#update_score changes in the future this will be a noop, which is whay I added the positive case in the previous example (expect when we rescore that the score changes, and that the blackbox is called).